File: app/src/main/java/me/srikavin/quiz/view/QuizDetailFragment.java
package me.srikavin.quiz.view;

import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.squareup.picasso.Picasso;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import me.srikavin.quiz.R;
import me.srikavin.quiz.view.game.GameActivity;
import me.srikavin.quiz.viewmodel.QuizDetailViewModel;

public class QuizDetailFragment extends Fragment {

    private QuizDetailViewModel mViewModel;

    public static QuizDetailFragment newInstance(String id) {
        QuizDetailFragment fragment = new QuizDetailFragment();
        Bundle args = new Bundle();
        args.putString("id", id);
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.quiz_detail_fragment, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        Toolbar toolbar = getView().findViewById(R.id.toolbar);
        ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mViewModel = ViewModelProviders.of(this).get(QuizDetailViewModel.class);

        final CollapsingToolbarLayout collapsingToolbar = getView().findViewById(R.id.collapsing_toolbar);
        collapsingToolbar.setTitle("title testing");
        Picasso.get().load((String) null)
                .placeholder(new ColorDrawable(getResources().getColor(R.color.colorSecondaryLight, null)))
                .into((ImageView) getView().findViewById(R.id.image));

        assert getArguments() != null;

        String id = getArguments().getString("id");
        assert id != null;


        mViewModel.getQuizByID(id).observe(this, quiz -> {
            if (quiz != null) {
                collapsingToolbar.setTitle(quiz.title);
                if (quiz.coverImage != null) {
                    Picasso.get().load(quiz.coverImage).into((ImageView) getView().findViewById(R.id.image));
                }
                TextView textView = getView().findViewById(R.id.quiz_detail_description);
                textView.setText(quiz.description == null ? "" : quiz.description);
            } else {
                Toast.makeText(getContext(), R.string.data_load_fail, Toast.LENGTH_LONG).show();
            }
        });

        getView().findViewById(R.id.quiz_detail_battle_fab).setOnClickListener((v) -> {
            Intent intent = new Intent(getActivity(), GameActivity.class);
            intent.putExtra("id", id);
            getActivity().startActivity(intent);
        });
    }

}

File: app/src/main/java/me/srikavin/quiz/view/QuizDetailFragment.kt
package me.srikavin.quiz.view

import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.google.android.material.appbar.CollapsingToolbarLayout
import com.squareup.picasso.Picasso
import me.srikavin.quiz.R
import me.srikavin.quiz.view.game.GameActivity
import me.srikavin.quiz.viewmodel.QuizDetailViewModel

class QuizDetailFragment : Fragment() {

    private var mViewModel: QuizDetailViewModel? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.quiz_detail_fragment, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        val toolbar = view!!.findViewById<Toolbar>(R.id.toolbar)
        (activity as AppCompatActivity).setSupportActionBar(toolbar)
        (activity as AppCompatActivity).supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        mViewModel = ViewModelProviders.of(this).get(QuizDetailViewModel::class.java)

        val collapsingToolbar = view!!.findViewById<CollapsingToolbarLayout>(R.id.collapsing_toolbar)
        collapsingToolbar.title = "title testing"
        Picasso.get().load(null as String?)
                .placeholder(ColorDrawable(resources.getColor(R.color.colorSecondaryLight, null)))
                .into(view!!.findViewById<View>(R.id.image) as ImageView)

        val id = arguments!!.getString("id")!!


        mViewModel!!.getQuizByID(id).observe(this, Observer { quiz ->
            if (quiz != null) {
                collapsingToolbar.title = quiz.title
                if (quiz.coverImage != null) {
                    Picasso.get().load(quiz.coverImage).into(view!!.findViewById<View>(R.id.image) as ImageView)
                }
                val textView = view!!.findViewById<TextView>(R.id.quiz_detail_description)
                textView.text = if (quiz.description == null) "" else quiz.description
            } else {
                Toast.makeText(context, R.string.data_load_fail, Toast.LENGTH_LONG).show()
            }
        })

        view!!.findViewById<View>(R.id.quiz_detail_battle_fab).setOnClickListener {
            val intent = Intent(activity, GameActivity::class.java)
            intent.putExtra("id", id)
            activity!!.startActivity(intent)
        }
    }

    companion object {
        fun newInstance(id: String): QuizDetailFragment {
            val fragment = QuizDetailFragment()
            val args = Bundle()
            args.putString("id", id)
            fragment.arguments = args
            return fragment
        }
    }

}
